Can't build testopt

Besides #98, when I try to build `testopt` with `ninja tests`, I get:

```
$ ninja tests
[1/1] Linking C executable test/testopt
FAILED: : && /usr/lib64/ccache/cc  -O2 -g -DNDEBUG  -Wl,--no-undefined test/CMakeFiles/testopt.dir/testfuncs.c.o test/CMakeFiles/testopt.dir/__/util/timer.c.o test/CMakeFiles/testopt.dir/__/util/mt19937ar.c.o  -o test/testopt  -rdynamic libnlopt.so.0.9.0 -lm -Wl,-rpath,/path/to/nlopt && :
/usr/lib/gcc/x86_64-redhat-linux/4.9.2/../../../../lib64/crt1.o: In function `_start':
(.text+0x20): undefined reference to `main'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

This is because NLopt explicitly does not enable C++, causing CMake to treat the source file containing `main` (`testopt.cpp`) as a generic text file rather than a source file to be compiled. (Note that the list of objects does not include `testopt.cpp.o` as it should.)

Obvious fix that may or may not be an acceptable approach:

```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 75d6029..7e504eb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,7 +18,7 @@ if (NOT DEFINED CMAKE_BUILD_TYPE)
   set(CMAKE_BUILD_TYPE Release CACHE STRING "Build type")
 endif ()
 
-project (nlopt C)
+project (nlopt C CXX)
 
 list (APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
 
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,15 +15,15 @@
 cmake_minimum_required (VERSION 2.8.5)
 
 if (NOT DEFINED CMAKE_BUILD_TYPE)
-  set(CMAKE_BUILD_TYPE Release CACHE STRING "Build type")
+  set (CMAKE_BUILD_TYPE Release CACHE STRING "Build type")
 endif ()
 
-project (nlopt C)
+project (nlopt)
 
 list (APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
 
 option (WITH_CXX "enable cxx routines" OFF)
-option (BUILD_SHARED_LIBS "Build NLOPT as a shared library" ON)
+option (BUILD_SHARED_LIBS "Build NLopt as a shared library" ON)
 option (BUILD_PYTHON "build python bindings" ON)
 option (BUILD_OCTAVE "build octave bindings" ON)
 option (BUILD_MATLAB "build matlab bindings" ON)
@@ -33,10 +33,6 @@ option (USE_SWIG "use SWIG to build bindings" ON)
 set (NLOPT_SUFFIX)
 if (WITH_CXX)
   set (NLOPT_SUFFIX _cxx)
-endif ()
-
-if (WITH_CXX OR BUILD_PYTHON OR BUILD_GUILE OR BUILD_OCTAVE)
-  enable_language (CXX)
 endif ()
 
 include (GNUInstallDirs)

